cuqi/distribution/_modifiedhalfnormal.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+import scipy.stats as sps
+import scipy.special as special
+from cuqi.distribution import Distribution
+from cuqi.utilities import force_ndarray
+
+class ModifiedHalfNormal(Distribution):
+    """
+    Represents a modified half-normal (MHN) distribution, a three-parameter family of distributions generalizing the Gamma distribution.
+    The distribution is continuous with pdf 
+    f(x; alpha, beta, gamma) propto x^(alpha-1) * exp(-beta * x^2 + gamma * x)
+
+    The MHN generalizes the half-normal distribution, because
+    f(x; 1, beta, 0) propto exp(-beta * x^2)
+
+    The MHN generalizes the gamma distribution because
+    f(x; alpha, 0, -gamma) propto x^(alpha-1) * exp(- gamma * x)
+
+    Reference:
+    [1] Sun, et al. "The Modified-Half-Normal distribution: Properties and an efficient sampling scheme." Communications in Statistics-Theory and Methods
+    
+    Parameters
+    ----------
+    alpha : float
+        The polynomial exponent parameter of the MHN distribution. Must be positive.
+
+    beta : float
+        The quadratic exponential parameter of the MHN distribution. Must be positive.
+
+    gamma : float
+        The linear exponential parameter of the MHN distribution.
+
+    """
+    def __init__(self, alpha, beta, gamma, is_symmetric=False, **kwargs):
+        # Init from abstract distribution class
+        super().__init__(is_symmetric=is_symmetric,**kwargs) 
+
+        self.alpha = alpha
+        self.beta = beta
+        self.gamma = gamma
+
+    def logpdf(self, x): # Unnormalized
+        return (self.alpha - 1)*np.log(x) - self.beta * x * x + self.gamma * x
+
+
+    def _MHN_sample_gamma_proposal(self, alpha, beta, gamma, delta = None):
+        if delta is None:

```

Comment:
Please add docstring, even to private methods, that explain what the method does.